src/lib/packager/product_patterns.rb

```diff
@@ -0,0 +1,141 @@
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# ------------------------------------------------------------------------------
+#
+
+require "yast"
+
+module Packager
+
+  # Evaluate the default patterns for the currently selected products
+  class ProductPatterns
+    include Yast::Logger
+
+    def initialize
+      Yast.import "Pkg"
+    end
+
+    # Find the default patterns for all selected products.
+    # @note The package management and the products needs to be initialized and
+    #   selected *before* using this method.
+    # @return [Array<String>] pattern names
+    def names
+      @names ||= find
+    end
+
+    # Select the default patterns to installation
+    # @note The package management and the products needs to be initialized and
+    #   selected *before* using this method.
+    # @return [Boolean] true if all patterns were successfuly selected,
+    #   false otherwise
+    def select
+      # use map + all? to try installing *all* patterns, plain all? would stop
+      # at the first failure
+      names.map { |p| Yast::Pkg.ResolvableInstall(p, :pattern) }.all?
+    end
+
+  private
+
+    # Find the default patterns for all selected products.
+    # @return [Array<String>] pattern names
+    def find
+      products = Yast::Pkg.ResolvableProperties("", :product, "")
+      products.select! { |product| product["status"] == :selected }
+      products.map! { |product| product["name"] }
```

Comment:
There can be several versions of the same product in the list but only one of them can be in `:selected` state. There should be no duplicates.